include/vkd3d_vk_includes.h

```diff
@@ -51,6 +51,7 @@ typedef struct D3D12_CUBIN_DATA_HANDLE
     UINT32 blockX;
     UINT32 blockY;
     UINT32 blockZ;
+    UINT32 smemSize;
```

Comment:
This is an ABI break and needs to be a new struct ala `D3D12_CUBIN_DATA_HANDLE1`

Reply:
This got me thinking, an alternative way of doing this would be to add a new function like `LaunchCubinShaderWithSmem` and do `ID3D12GraphicsCommandListExt1` instead of extending the device further.

This would better fit with `VK_NVX_binary_import`, as well as the `NvAPI_D3D12_LaunchCuKernelChain` and `NvAPI_D3D12_LaunchCuKernelChainEx` functions. It seems like that's probably a better goal to hit here (especially given @Joshua-Ashton's comment on wanting to limit future interface iterations). Just extending the command-list to take in smem + pParams should help us avoid future changes around the launch interface.

Does that seem like a reasonable move to you? I think with this ABI breakage we would need to bump the command-list interface anyways, so I prefer this new alternative to what I had previously planned.

Reply:
Extending command list interface is fine too.

Reply:
I've implemented that. The new `LaunchCubinShaderEx` function introduced should also cover @Joshua-Ashton's initial request to plumb `pParams` to avoid further iterations on this interface.